test/integration/targets/unarchive/tasks/test_zip_nonascii.yml

```diff
@@ -0,0 +1,30 @@
+- set_fact:
+    dest: '{{ remote_tmp_dir }}/test-unarchive-nonascii-zip'
+
+- name: create our zip unarchive destination
+  file:
+    path: '{{ dest }}'
+    state: directory
+
+- name: unarchive a zip file with non-ascii filenames
+  unarchive:
+    src: 'test-zip-nonascii.zip'
+    dest: '{{ dest }}'
+    remote_src: no
+  register: unarchive_non_ascii_01
```

Comment:
Can you add a repeat task to check it's successful and changed == false to cover the KeyError reproducer from #49408?